dynamiqs/qarrays/qarray.py

```diff
@@ -45,7 +45,7 @@ class QArray(eqx.Module):
     def __check_init__(self):
         # === ensure dims is a tuple of ints
         if not isinstance(self.dims, tuple) or not all(
-            isinstance(d, int) for d in self.dims
+            isinstance(d, int) or np.issubdtype(d.dtype, int) for d in self.dims
```

Comment:
We should only support `tuple[int, ...]` (otherwise we have to change the typing everywhere). IMO the calling method should convert `np.int` object to proper `int`.